### PR TITLE
Micson test wip

### PR DESCRIPTION
**Pull request by micson-us**
**Description:** Updated workflow to run coverage test before building the image. Workflow will fail if coverage report is below 80%.

**Issue:** Provide the URL for a corresponding issue/work ticket, if applicable.

**Dependencies:** List dependencies on open pull requests, other issue/work tickets, etc.

**Merge Deadline:** Indicate the merge deadline date.

**Pre-Merge Checklist:**

- [x] I have ensured this code follows the style guidelines of this project.
- [x] I have performed a comprehensive self-review of the code.
- [x] I have included comments clarifying hard-to-understand areas of the code.
- [x] I have updated the documentation corresponding to my change(s).
- [x] I have confirmed the code executes properly on my local environment.
- [x] I have confirmed all new/modified data models are structured correctly.
- [x] I have included all necessary database migration files.
- [x] I have verified that all new and existing tests pass.
- [x] I have received approval from the requisite number of reviewers.
- [x] I have squashed all working commits into as few final commits as reasonably possible.
- [x] I have confirmed Github reports all automated tests/checks are passing.

**Post-Merge Checklist:**

- [ ] Delete working branch @ Github repository
- [ ] Pull latest master branch to local repository
- [ ] Delete working branch @ local repository

### CODE REVIEW

File: .github/workflows/cd-workflow.yml [hunk 1]
```diff
@@ -7,9 +7,53 @@ on:
...
+        pip install -r ./requirements.txt
+# Start running test scripts and generate the coverage report
+    - name: Generate Coverage Report
+      env:
+        GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+      run: |
+        pip install pytest
+        pip install pytest-cov
+        echo "Cloning private repository"
+        git clone https://micson-us:${{ secrets.ACCESS_TOKEN }}@github.com/OpenLXP/openlxp-private.git
```
Comment (mattdrayer) [inline .github/workflows/cd-workflow.yml:37]: What is `micson-us` in this context?  Could it be something more like `openlxp-host`?

File: .github/workflows/cd-workflow.yml [hunk 2]
```diff
@@ -20,10 +64,13 @@ jobs:
        DOCKER_USER: ${{ secrets.DOCKER_USER }}
        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
       run: |
+       echo "Authenticate to Docker"
        docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
     - name: Docker build
```
Comment (mattdrayer) [inline .github/workflows/cd-workflow.yml:69]: Nit: docker -> Docker

**Review by mattdrayer — APPROVED** [review]
Other than one question about a hostname it LGTM!

File: .github/workflows/cd-workflow.yml (hunk 1 above)
Comment (micson-us) [reply .github/workflows/cd-workflow.yml:37]: Hi Matt - that is my username. I will create a service account to run these workflows. openlxp-host is a great name for it!

**Review by jdiouf — APPROVED** [review]

**Pull request merged**